bundle-workflow/src/test_workflow/utils/build_manifest_provider.py

```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import os
+
+from aws.s3_bucket import S3Bucket
+from manifests.build_manifest import BuildManifest
+
+
+class BuildManifestProvider:
+    @staticmethod
+    def get_build_manifest_relative_location(build_id, opensearch_version, architecture):
+        return f"builds/{opensearch_version}/{build_id}/{architecture}/manifest.yml"
+
+    @staticmethod
+    def load_manifest(bucket_name, build_id, opensearch_version, architecture):
+        local_path = str(os.getcwd())
```

Comment:
This should be a temporary file name or if streaming is supported no file at all.

Reply:
the `S3Bucket.download_file` method requires a local destination as input.